### PR TITLE
[RISCV] Bump Pointer Masking extension version

### PR DESCRIPTION
**Pull request by michaelmaitland**
These extensions had their version number bumped and still experimental (under public review). I didn't see anything in the [commit history](https://github.com/riscv/riscv-j-extension/commits/master/) since #79929 that would warrant a change to the implementation of pointer masking in the compiler.

### CODE REVIEW

File: llvm/docs/RISCVUsage.rst [hunk 1]
```diff
@@ -263,7 +263,7 @@ LLVM supports (to various degrees) a number of experimental extensions.  All exp
 The primary goal of experimental support is to assist in the process of ratification by providing an existence proof of an implementation, and simplifying efforts to validate the value of a proposed extension against large code bases.  Experimental extensions are expected to either transition to ratified status, or be eventually removed.  The decision on whether to accept an experimental extension is currently done on an entirely case by case basis; if you want to propose one, attending the bi-weekly RISC-V sync-up call is strongly advised.
 
 ``experimental-ssnpm``, ``experimental-smnpm``, ``experimental-smmpm``, ``experimental-sspm``, ``experimental-supm``
-  LLVM implements the `v0.8.1 draft specification <https://github.com/riscv/riscv-j-extension/blob/master/zjpm-spec.pdf>`__.
+  LLVM implements the `v1.0.0-rc2 specification <https://github.com/riscv/riscv-j-extension/blob/master/zjpm-spec.pdf>`__.
```
Comment (topperc) [inline llvm/docs/RISCVUsage.rst:266]: Please use the versioned link here https://github.com/riscv/riscv-j-extension/releases

**Review by jaidTw — COMMENTED** [review]
Need to update `clang/test/Preprocessor/riscv-target-features.c`

**Review by jaidTw — APPROVED** [review]
LGTM

File: llvm/docs/ReleaseNotes.rst [hunk 2]
```diff
@@ -184,6 +184,7 @@ Changes to the RISC-V Backend
 * B (the collection of the Zba, Zbb, Zbs extensions) is supported.
 * Added smcdeleg, ssccfg, smcsrind, and sscsrind extensions to -march.
 * ``-mcpu=syntacore-scr3-rv32`` and ``-mcpu=syntacore-scr3-rv64`` were added.
+* Ssnpm, Smnpm, Smmpm, Sspm, and Supm are bumped to version 1.0.0
```
Comment (topperc) [inline llvm/docs/ReleaseNotes.rst:187]: Please update line 174 instead.

**Review by topperc — APPROVED** [review]
LGTM

**Pull request merged**